### DISCUSSION TITLE
Avoid Wyze app / cloud / account altogether?

### DISCUSSION THREAD

**Discussion by pjv** [gtxaspec/wz_mini_hacks#144] [Q&A]
Excuse the noob post but I have a brand new wyzecam v3 that has yet to be powered up and I'm wondering if i can use wz_mini_hacks to get the camera on my LAN and broadcasting via RTSP without installing the wyze app on my phone, setting up a wyze account, etc. Is this possible?

**Comment by virmaior** [comment]
See https://github.com/gtxaspec/wz_mini_hacks/issues/38 and https://github.com/gtxaspec/wz_mini_hacks/issues/58 
I think the current consensus is:
1. Need the app to initially setup the camera
2. Then you can disconnect it from the internet.
3. To do LAN RTSP, you would need to downgrade the firmware to 4.36.2.5 via SD card, because later versions phone home and won't start without phoning home.
4. After that you can then install wz_mini and enabling RTSP

(correct decimal position)

**Reply by pjv** [reply]
Is there easy access to firmware 4.3.6.25 somewhere?

**Reply by virmaior** [reply]
https://github.com/gtxaspec/wz_mini_hacks/issues/38#issuecomment-1133774809

4.36.2.5  (https://download.wyzecam.com/firmware/v3/demo_wcv3_4.36.2.5.bin.zip) 
I had my decimal in the wrong place.

**Comment by endertable** [comment]
I did the following with success:

1.  Copied /configs/.user_config from another cam onto the SD card into the \wz_mini directory and named it usrconfig
2.  Copied wz_mini to SD card then populated the \wz_mini\etc\ssh\authorized_keys file (for ssh access)
3.  Created a custom script with the following  and placed on SD card.
4.  Added script to the bottom of the wz_mini.conf file to run as custom:


#!/bin/sh
if ! [ -f /configs/.user_config ]
then
  echo First time offline provisioning
  echo -n "mySID" > /configs/.wifissid
  echo -n "myPasswd" > /configs/.wifipasswd
  rm -f /configs/.wifipasswd.enc
  cp /media/mmc/wz_mini/usrconfig /configs/.user_config
  echo "1656630463" > /configs/.time_checkpoint
  reboot
fi


Obviously you will have to update the date manually and update /configs/.time_checkpoint (this is so you don't continue overwriting your SD card mp4 files if you have that turned on in the .user_config file).

This is a complete offline solution.

**Reply by pjv** [reply]
Wow, that's great info. Thank you very much. Only problem for me is this is my first wyzecam so I have no source for step 1.

**Reply by endertable** [reply]
I can send you a copy, unless you already did online provision.

**Reply by pardovot** [reply]
Is this still a valid solution?
As far as I can see, custom scripts will only run after a network connection is established, if a connection is not established it will never run this custom script. Am I missing something?

**Reply by endertable** [reply]
@pardovot yes, this is still a valid solution.  But to your point wz_mini has gone through a major overhaul in its structure since this solution was written.  Wyze firmware has also had some updates, so this is good for up to 4.36.9.139.  Also, the creator is not keen on having any of his programs writing to the /configs directory both for your safety and his sanity, so you will have to interject your own coding and script.

The basic gist to getting your camera up and connected  to your wifi without the app (can be placed in a startup script to run 1 time):

```
echo -n "yourWifiSid" > /configs/.wifissid
echo -n "yourWifiPasswd" > /configs/.wifipasswd
echo "1694734609" > /configs/.time_checkpoint

cat <<- EOF > /configs/.user_config
[NET]
bindOk=1

[SETTING]
indicator=2
watermark_flag=2
EOF
```

**Reply by pardovot** [reply]
Thanks for the reply, I actually am on 4.36.9.139, but still I can see in the logs that my custom script will not run since there is no connection, I don't know if maybe I should first completely reset to camera or something so the wifi settings are removed and then it won't try to connect to the wifi and will proceed to the custom scripts.

Also, I was trying to modify the .wifissid (I have the same password for all the wifi's I used), so I just do nano /configs/.wifissid, change the wifi ssid, save it and restart the wyze camera, but it still won't connect to the wifi set in the /configs folder, is there anything I'm missing? Can it even work?
Basically, I will use the camera with different routers in some cases, and I want to be able to easily switch a network.
Thanks.

**Reply by endertable** [reply]
Will it connect without wz_mini active?  Rename the /factory_t31_ZMC6tiIDQN on the root of the SD card to /_factory_t31_ZMC6tiIDQN, reboot. and see if it connects to network.  If not, a few questions:

Did you have this on a higher FW than 4.36.9.139 before?
Does this refuse to connect using the Wyze app also, or is that not an option?
Can you recreate the /configs/.wifipasswd and the /configs/.wifissid using the "echo" commands above vs nano?  Not sure if it makes a difference, but nano adds a NL to the end.

**Reply by pardovot** [reply]
So I didn't actually tried the custom script as that's generally not my purpose, but I did try to write to .wifissid without NL and it works great! The best part about it is that if I change the network to another the app would still work in case I have internet access on my new wifi, which is really fantastic.
Thanks a lot!

**Comment by gtxaspec** [comment]
be careful with the files in /configs/, some are generated unique for each camera.  If these files (that have keys), are deleted or corrupted, in my testing the cameras will never connect to the cloud server (they will fail authentication, as the keys are generated by the factory for each camera uniquely), either initially or ever again.

**Reply by endertable** [reply]
Was this a V3 camera that you can’t get back online, because if so Wyze usually stores a copy of .product_config on an unmounted partition

**Reply by gtxaspec** [reply]
multiple camera's that the flash chip has been replaced on... no backups of the files lol

**Comment by endertable** [comment]
Very true,  the above is for a complete offline solution only as the OP requested.  Do NOT ever mess with the .product_config file or it’s backup which has an MD5 hash in the filename.  And if you decide to put cam online, before you do, remove only the following files from /configs:

.wifi*
.user_config
.time_checkpoint

Then disable wz_mini temporarily by renaming the factory* file on the root of the SD card then reboot and use the Wyze app as if it’s a brand new cam.  After Wyze provisioning you can re enable wz_mini
